app/controllers/tool_groups_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ToolGroupsController < ApplicationController
+  before_action :authorize!
+
+  def index
+    render json: ToolGroup.all.order(name: :asc), status: :ok
+  end
+
+  def create
+    create_tool_group
+  rescue ActiveRecord::RecordInvalid => e
+    render json: {error: e.record.errors}, status: :unprocessable_entity
```

Comment:
so this error format isn't currently following the jsonapi spec, so our client parsers won't be able to parse it correctly.

relevant [examples](https://jsonapi.org/examples/#error-objects-basics) and the (spec)[https://jsonapi.org/format/#error-objects]

So, currently the errors are being returned like this:

```json
{
    "error": {
        "suggestions_weight": [
            "can't be blank"
        ],
        "name": [
            "has already been taken"
        ]
    }
}
```

When it should look more like:
```json
{
  "errors": [
    {
      "detail": "name has already been taken"
    },
    {
      "detail": "suggestions-weight can't be blank"
    }
  ]
}
```

There are other possible fields that can be included in the error objects, but the detail one is what we currently use on the admin UI to display any errors returned from the API

Reply:
It actually looks like if you raise an error, it will automatically be formatted as a jsonapi error correctly.

https://github.com/CruGlobal/mobile-content-api/blob/master/app/controllers/languages_controller.rb#L19

yields the following JSON:
```json
{
    "errors": [
        {
            "source": {
                "pointer": "/data/attributes/id"
            },
            "detail": "Code en already exists."
        }
    ]
}
```

Reply:
it looks like there is some common methods for rendering jsonapi errors that can probably be utilized:

https://github.com/CruGlobal/mobile-content-api/blob/master/app/controllers/application_controller.rb#L68-L86

Reply:
It should be fixed now